api/v1alpha1/ratelimitfilter_types.go

```diff
@@ -0,0 +1,164 @@
+}
+
+// +kubebuilder:object:root=true
+
+// RateLimitFilterList contains a list of RateLimitFilter resources.
+type RateLimitList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []RateLimitFilter `json:"items"`
+}
+
+// RateLimitFilterSpec defines the desired state of RateLimitFilter.
+// +union
+type RateLimitFilterSpec struct {
+	// Type decides the scope for the RateLimits.
+	// Valid RateLimitType values are:
+	//
+	// * "Global" - In this mode, the rate limits are applied across all Envoy proxy instances.
+	//
+	// +unionDiscriminator
+	Type RateLimitType `json:"type"`
+	// Global rate limit configuration.
+	//
+	// +optional
+	Global *GlobalRateLimit `json:"global,omitempty"`
+}
+
+// RateLimitType specifies the types of RateLimiting.
+// +kubebuilder:validation:Enum=Global
+type RateLimitType string
+
+const (
+	// GlobalRateLimitType allows the rate limits to be applied across all Envoy proxy instances.
+	GlobalRateLimitType RateLimitType = "Global"
+)
+
+// GlobalRateLimit defines the global rate limit configuration.
+type GlobalRateLimit struct {
+	// Rules are a list of RateLimit selectors and limits.
+	// Each rule and its associated limit is applied
+	// in a mutually exclusive way i.e. if multiple
+	// rules get selected, each of their associated
+	// limits get applied, so a single traffic request
+	// might increase the rate limit counters for multiple
+	// rules if selected.
+	//
+	// +kubebuilder:validation:MaxItems=16
+	Rules []RateLimitRule `json:"rules"`
+}
+
+// RateLimitRule defines the semantics for matching attributes
+// from the incoming requests, and setting limits for them.
+type RateLimitRule struct {
+	// ClientSelectors holds the list of select conditions to select
+	// specific clients using attributes from the traffic flow.
+	// All individual select conditions must hold True for this rule
+	// and its limit to be applied.
+	// If this field is empty, it is equivalent to True, and
+	// the limit is applied.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=8
+	ClientSelectors []RateLimitSelectCondition `json:"clientSelectors,omitempty"`
+	// Limit holds the rate limit values.
+	// This limit is applied for traffic flows when the selectors
+	// compute to True, causing the request to be counted towards the limit.
+	// The limit is enforced and the request is ratelimited, i.e. a response with
+	// 429 HTTP status code is sent back to the client when
+	// the selected requests have reached the limit.
+	Limit RateLimitValue `json:"limit"`
+}
+
+// RateLimitSelectCondition specifies the attributes within the traffic flow that can
+// be used to select a subset of clients to be ratelimited.
+// All the individual conditions must hold True for the overall condition to hold True.
+type RateLimitSelectCondition struct {
+	// Headers is a list of all header matches that must be matched
+	// for the overall select condition to hold True.
```

Comment:
I find it difficult to parse this explanation. Would the following explanation be valid?
```
Headers is a list of request headers to match. Multiple header values are ANDed together, meaning, a request MUST match all the specified headers.

Reply:
sure sounds clearer, will add